src/lib.rs

```diff
@@ -324,6 +324,43 @@ pub struct WalletOpts {
     #[cfg(feature = "esplora")]
     #[structopt(flatten)]
     pub esplora_opts: EsploraOpts,
+    #[cfg(feature = "compact_filters")]
+    #[structopt(flatten)]
+    pub compactfilter_opts: CompactFilterOpts,
+}
+
+/// Compact Filter options
+///
+/// Compact filter peer information used by [`OnlineWalletSubCommand`]s.
+#[cfg(feature = "compact_filters")]
+#[derive(Debug, StructOpt, Clone, PartialEq)]
+pub struct CompactFilterOpts {
+    /// Sets the SOCKS5 proxy for the Electrum client
+    #[structopt(name = "CPROXY_SERVER:PORT", long = "cmpct_proxy")]
```

Comment:
Since both electrum and cbf take an optional proxy I think it'd be clearer to move the proxy options from ElectrumOpts (proxy, retries, timeout) ,and from CompactFilterOpts (proxy, user, passwd) and combine into a new ProxyOpts struct. Then add a ProxyOpts field to WalletOpts and enable it when electrum or compact_filters features are enabled. The specific unused proxy opts can be ignored, possibly with a warning.. ie. the user supplies a  user/password for the electrum client.  Also to reduce the number of opts.. I think you could combine user and passwd into proxy_auth that takes a "PROXY_USER:PASSWD". Ideally in the future we'll be able to get all options supported for any blockchain client.

Reply:
Yes, thanks this makes sense. 

Going with a dedicated `ProxyOpts` with `PROXY_USER:PASSWD` pattern.